src/components/TodoApp/TodoApp.jsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState, useEffect } from 'react';
+import './style.css'
+import axios from 'axios';
+import AddTodo from '../AddTodo/AddTodo';
+import Todos from '../Todos/Todos';
+
+function TodoApp() {
+  const [allTasks, setAllTasks] = useState([]);
+  const url = 'http://localhost:8080';
+
+  useEffect(() => {
+    getAllTasks();
+  }, []);
+
+  const getAllTasks = async() => {
+    try {
+      const resp = await axios.get(`${url}/allTasks`);
+      setAllTasks(resp.data);
+    } catch (error) {
+      alert('Ошибка в получении всех задач')
+    };
+  };
+
+  const addTask = async(userInput) => {
+    if (userInput.trim() === '') {
+      userInput = ''
+      return;
+    };
+    try {
+      const resp = await axios.post(`${url}/createTask`, {
+        text: userInput,
+        isCheck: false
+      })
+      setAllTasks([...allTasks, resp.data]);
+    } catch (error) {
+      alert('Ошибка в добавлении задачи');
+    };
+  };
+
+  const delelteTask = async(_id) => {
+    try {
+      await axios.delete(`${url}/deleteTask/?id=${_id}`, {
+        data: {_id},
+        headers: {
+          'Content-type': 'application/json'
+        }
+      });
+      setAllTasks([...allTasks.filter(task => task._id !== _id)]);
+    } catch (error) {
+      alert('Ошибка в удалении задачи')
+    };
+  };
+
+  const completedTask = async(_id,) => {
+    try {
+      const findTask = allTasks.find(item => item._id === _id);
+      const resp = await axios.patch(`${url}/updateTaskCheck`, {
+        _id: _id,
+        isCheck: !findTask.isCheck
+      });
+      
+      const sortArr = resp.data.sort((a, b) => a.isCheck > b.isCheck ? 1 : a.isCheck < b.isCheck ? -1 : 0);
```

Comment:
what kind of record is this? resp.data.sort

And you forgot about mutations, just reassigned the array